商品詳細機能の実装

Yusuke0107 opened the pull request:
# What
商品詳細機能の実装

# Why
商品の詳細画面を表示させる為
ログイン状況によって挙動を変える為

# Gyazo
[・ログイン状態且つ、自身が出品した販売中商品の商品詳細ページへ遷移した動画
](https://gyazo.com/9031f50966605a9e57861b7ca2628ba0)
[・ログイン状態且つ、自身が出品していない販売中商品の商品詳細ページへ遷移した動画
](https://gyazo.com/df1ee011780a14cb22fee25aed1ff412)
・ログイン状態で、売却済み商品の商品詳細ページへ遷移した動画（現段階で商品購入機能の実装が済んでいる場合） 
※商品購入機能未実装の為無し
[・ログアウト状態で、商品詳細ページへ遷移した動画
](https://gyazo.com/9f14369242eeb43759ca26aec054e142)

Yusuke0107 committed:
商品詳細機能の実装

shodai-ito commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,114 @@
+<div class="item-show">
+  <div class="item-box">
+    <h2 class="name">
+      <%= "商品名" %>
```
商品名が設定されていないため、記述をお願いします！

shodai-ito commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,114 @@
+    </div>
+    <div class="item-price-box">
+      <span class="item-price">
+        ¥ 999,999,999
```
商品の値段が設定されていないため、記述をお願いします！

shodai-ito commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,114 @@
+        ¥ 999,999,999
+      </span>
+      <span class="item-postage">
+        <%= "配送料負担" %>
```
配送料負担が設定されていないため、記述をお願いします！

shodai-ito commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,114 @@
+    <%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
+
+    <div class="item-explain-box">
+      <span><%= "商品説明" %></span>
```
商品説明が設定されていないため、記述をお願いします！

shodai-ito commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,114 @@
+    </a>
+  </div>
+  <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
+  <a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
```
商品のカテゴリー名が設定されていないため、記述をお願いします！

shodai-ito commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,114 @@
+    <%# //商品が売れていない場合はこちらを表示しましょう %>
+
+
+    <%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
```
「〜しましょう」で終わる実装案内コメントは実装完了後に削除しましょう。
企業提出時に実装案内コメントが残っていないようにするためです。

shodai-ito commented on app/controllers/items_controller.rb:
```diff
@@ -1,5 +1,7 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!, only: [:new]
+  before_action :set_item, only: [:edit, :show]
```
editアクションに関する記述は次回の実装内容であるため、現段階ではコメントアウトか削除をお願いいたします！

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！
プルリクエストの内容を確認させていただきました。

修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```


ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞよろしくおねがいします。

Yusuke0107 committed:
商品詳細機能の実装 修正①

①商品名、値段、配送料負担、商品説明、カテゴリー名が設定されていない為記述
②実装案内コメントが残っていた為、削除
③editアクションに関する記述は次回の実装内容であるため、削除

wad-sho commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,102 @@
+
+  <% if user_signed_in? && current_user.id != @item.user_id %>
+    <%= link_to "購入画面に進む", "#" ,class:"item-red-btn"%>
+  <% end %>
```
リファクタリングをしましょう！
以下の形式で記述すると、同じ挙動で、記述量も減り、可読性の向上にも繋がるのでご参考にいただけますと幸いでございます。

```ruby
if "ログインしていれば"
  if "現在ログインしているユーザーが商品の出品者であれば"
    商品の編集
    削除
  else
    購入画面に進む
  end
end
```

wad-sho reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。

コメントいたしましたので、ご確認お願いします。

修正の後、再度レビューのご依頼よろしくお願いいたします！

Yusuke0107 committed:
商品詳細機能の実装 修正②

 ① <% if user_signed_in? && current_user.id != @item.user_id %>の記述を記述量が減り、可読性を上げるものに修正

motomuun commented:
再度レビューのご依頼ありがとうございます。

実装内容を確認させていただきました。
**LGTMとさせていただきます。実装お疲れ様でした！**

このブランチをマージし、masterブランチ（もしくはmainブランチ）でプルしましょう。
内容が反映されていることを確認し、次の実装・学習に入りましょう！
引き続き応援しております！
